hail/src/main/scala/is/hail/types/physical/PCanonicalArray.scala

```diff
@@ -85,6 +99,11 @@ final case class PCanonicalArray(elementType: PType, required: Boolean = false)
     _elementsOffset(length)
   }
 
+  def missingBytesOffset: Long = lengthHeaderBytes
```

Comment:
> This is a misleading name. We pervasively use the word offset to mean address (we should stop doing that).

Agreed, but this is unfortunately also consistent with `elementsOffset`. I'm very tempted to just go through and fix all the naming, at least in `PCanonicalArray`, but maybe after this PR.

What I'm trying to do here is draw some boundaries around what details of the memory layout consumers may depend on. So far I've done that just by making some things private, but it needs to be described in text too. My current thought is: there is an integer length, a byte array of missingness bits padded to a multiple of 8 bytes, and an elements array. The start of each of the three sections is 8-byte aligned. Each section occupies a contiguous slice of memory. Consumers may access the start address and length of each section, but may not depend on their relative position, such as their order or the distance in between them.

This line was because the length of the header shouldn't be accessible, but the start of the missing bytes should. The naming was to be consistent with the other section start, `elementsOffset`, which I agree is terrible and needs to be changed.